src/path.js

```diff
@@ -27,7 +27,7 @@ const validate = require('./validate')();
  *
  * @type {RegExp}
  */
-const RESOURCE_PATH_RE = /^projects\/([^/]*)\/databases\/([^/]*)(?:\/documents\/)?(.*)$/;
+const RESOURCE_PATH_RE = /^projects\/([^/]*)\/databases\/([^/]*)(?:\/documents\/)?([\s\S]*)$/;
```

Comment:
While more correct, this new regex is less readable. :p Googling for \s\S does not yield any useful results. Consider adding a comment.

Reply:
It's the JS equivalent of the /s flag (which matches newlines in most RegExp parsers). I added a comment.